examples/bond/scripts/run/common/common.js

```diff
@@ -68,6 +68,209 @@ async function optInTx (deployer, managerAcc, lsig, assetIndex) {
   await executeTransaction(deployer, optInTx);
 }
 
+/**
+ * Returns group transaction for buy
+ * @param buyer buyer account
+ * @param issuerLsig issuer logic signature
+ * @param amount amount of bonds
+ * @param algoAmount amount of algo
+ * @param appID application index
+ * @param bondID bond index
+ */
+function buyTx (buyer, issuerLsig, amount, algoAmount, appID, bondID) {
+  return [
+    // Algo transfer from buyer to issuer
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: buyer,
+      toAccountAddr: issuerLsig.address(),
+      amountMicroAlgos: algoAmount,
+      payFlags: { totalFee: 2000 }
+    },
+    // Bond token transfer from issuer's address
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: issuerLsig.address(),
+      lsig: issuerLsig,
+      toAccountAddr: buyer.addr,
+      amount: amount,
+      assetID: bondID,
+      payFlags: { totalFee: 0 }
+    },
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: buyer,
+      appID: appID,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:buy']
+    }
+  ];
+}
+
+/**
+ * Returns issue group transaction
+ * @param creatorAccount creator acccount
+ * @param issuerLsig issuer logic signature
+ * @param appID application index
+ * @param bondID bond index
+ */
+function issueTx (creatorAccount, issuerLsig, appID, bondID) {
+  return [
+    // Bond asa transfer to issuer's address
+    {
+      type: types.TransactionType.TransferAsset,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      toAccountAddr: issuerLsig.address(),
+      amount: 1e6,
+      assetID: bondID,
+      payFlags: { }
+    },
+    // call to bond-dapp
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: creatorAccount,
+      appID: appID,
+      payFlags: {},
+      appArgs: ['str:issue']
+    }
+  ];
+}
+
+/**
+ * Returns create dex transaction
+ * @param managerAccount App manager account
+ * @param appID application index
+ * @param issuerLsig issuer logic signature
+ * @param dexLsig dex logic signature
+ * @param creatorAccount bond token creator account
+ * @param assetAmount balance of oldBond tokens in issuer address
+ * @param newBondID new bond token index
+ * @param oldBondID old bond token index
+ * @param total total number of bonds issued to buyers
+ * @param couponValue value of coupon
+ */
+function createDexTx (
+  managerAccount, appID, issuerLsig,
+  dexLsig, creatorAccount, assetAmount,
```

Comment:
let's rename `asseetAmount` to `unsoldBonds`.

Reply:
ok